core/src/main/java/org/mule/runtime/core/api/MuleEvent.java

```diff
@@ -40,6 +42,40 @@
   int TIMEOUT_NOT_SET_VALUE = Integer.MIN_VALUE;
 
   /**
+   * @return the context applicable to all events created from the same root {@link MuleEvent} from a {@link MessageSource}.
+   */
+  MessageExecutionContext getExecutionContext();
+
+  /**
+   * Returns the correlation metadata of this message. See {@link Correlation}.
+   * 
+   * @return the correlation metadata of this message.
+   */
+  Correlation getCorrelation();
+
+  /**
+   * The correlation Id can be used by components in the system to manage message relations.
+   * <p>
+   * The id is associated with the message using the underlying transport protocol. As such not all messages will support the
+   * notion of a id i.e. tcp or file. In this situation the correlation Id is set as a property of the message where it's up to
+   * developer to keep the association with the message. For example if the message is serialised to xml the id will be available
+   * in the message.
+   *
+   * @return the id for this message or null value if one hasn't been set
+   */
+  String getCorrelationId();
+
+  /**
+   * @return {@code true} if the source of this event has set a correlation id (see {@link #getCorrelationId()}).
+   */
+  boolean hasSourceCorrelation();
```

Comment:
I don't think this is needed, from event perspective it should be transparent if source set a custom id or not.